.circleci/config.yml

```diff
@@ -10,70 +11,73 @@ jobs:
       tag: "17.0"
     steps:
       - checkout
-      - run: mkdir -p ~/build-artifacts
+      - run: mkdir -p ./build-artifacts
       - gradle/with_cache:
           steps:
             - run: ./gradlew build -x test
       - run:
-          name: Copy JARs to build-artifacts
-          command: |
-            shopt -s globstar
-            cp **/build/libs/*.jar ~/build-artifacts
+          name: Show the JARs
+          command: ls -l build-artifacts
       - persist_to_workspace:
-          root: ~/
+          root: ./
           paths:
             - build-artifacts
 
   smoke_test:
-    docker:
-      - image: cimg/base:stable
-    environment:
-      AGENT_JAR_GLOB: $HOME/build-artifacts/honeycomb-opentelemetry-javaagent*-all.jar
-      AGENT_ONLY_APP_JAR_GLOB: $HOME/build-artifacts/spring-agent-only-?.?.?.jar
-      AGENT_MANUAL_APP_JAR_GLOB: $HOME/build-artifacts/spring-agent-manual-?.?.?.jar
+    machine:
+      image: ubuntu-2004:202107-02
```

Comment:
docker -> machine executor. Includes an up-to-date version of Docker, Docker-Compose, and such which runs local to the virtual machine and lets us do volume mounts from the test host and loopback port forwards.